Sources/Caching/FileRepository.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  FileRepository.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// A file cache
+@_spi(Internal) public class FileRepository: FileRepositoryType, @unchecked Sendable {
+    let networkService: SimpleNetworkServiceType
+
+    private let store = KeyedDeferredValueStore<InputURL, OutputURL>()
+    private let fileManager: LargeItemCacheType
+
+    /// Create a file repository
+    /// - Parameters:
+    ///   - networkService: A service capable of fetching data from a URL
+    ///   - fileManager: A service capable of storing data and returning the URL where that stored data exists
+    @_spi(Internal) public init(
+        networkService: SimpleNetworkServiceType = URLSession.shared,
+        fileManager: LargeItemCacheType = FileManager.default
+    ) {
+        self.networkService = networkService
+        self.fileManager = fileManager
+    }
+
+    /// Prefetch files at the given urls
+    /// - Parameter urls: An array of URL to fetch data from
+    @_spi(Internal) public func prefetch(urls: [InputURL]) {
+        for url in urls {
+            Task { [weak self] in
+                try await self?.getCachedURL(for: url)
+            }
+        }
+    }
+
+    /// Create and/or get the cached file url
+    /// - Parameters:
+    ///   - url: The url for the remote data to cache into a file
+    @_spi(Internal) public func getCachedURL(for url: InputURL) async throws -> OutputURL {
```

Comment:
I think I would expect a method with the name of `getCachedURL` to get not "create or get" 🤔  I think based on that name, I would expect it to be a non-async method that doesn't have a `downloadFile` in it.

But that being said... I think this implementation makes sense! So I think if we could get a name that is matches doc of "Create and/or get the cached file url" would be 💯